sklearn/cluster/_bicluster.py

```diff
@@ -139,22 +140,22 @@ def _svd(self, array, n_components, n_discard):
                                       **kwargs)
 
         elif self.svd_method == 'arpack':
-            u, _, vt = svds(array, k=n_components, ncv=self.n_svd_vecs)
+            random_state = check_random_state(self.random_state)
+            v0 = _init_arpack_v0(min(array.shape), random_state)
+            u, _, vt = svds(array, k=n_components, ncv=self.n_svd_vecs, v0=v0)
             if np.any(np.isnan(vt)):
                 # some eigenvalues of A * A.T are negative, causing
                 # sqrt() to be np.nan. This causes some vectors in vt
                 # to be np.nan.
                 A = safe_sparse_dot(array.T, array)
-                random_state = check_random_state(self.random_state)
-                # initialize with [-1,1] as in ARPACK
-                v0 = random_state.uniform(-1, 1, A.shape[0])
+                # We have to renitialize v0 differently because the shape
+                # here can be different from the previous init
+                v0 = _init_arpack_v0(A.shape[0], random_state)
```

Comment:
Can we revert this change so codecov is happy? (This is already doing the correct thing, but not using the new helper function)

We can work on a test for updating this in another PR.